test/ValidatorSet/ValidatorSet.test.ts

```diff
@@ -356,18 +354,134 @@ describe("ValidatorSet", function () {
+            .connect(this.signers.governance)
+            .addToWhitelist([this.signers.validators[0].address, this.signers.validators[1].address])
+        ).to.not.be.reverted;
+
+        await expect(
+          validatorSet
+            .connect(this.signers.governance)
+            .addToWhitelist([this.signers.validators[0].address, this.signers.validators[1].address])
+        ).to.be.reverted;
+
+        expect((await validatorSet.validators(this.signers.validators[0].address)).status).to.be.equal(
+          VALIDATOR_STATUS.Whitelisted
+        );
+        expect((await validatorSet.validators(this.signers.validators[1].address)).status).to.be.equal(
+          VALIDATOR_STATUS.Whitelisted
+        );
       });
 
       it("should be able to remove from whitelist", async function () {
         const { validatorSet } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
 
+        await expect(
+          validatorSet
+            .connect(this.signers.governance)
+            .addToWhitelist([this.signers.validators[1].address, this.signers.validators[3].address])
+        ).to.not.be.reverted;
+
         await expect(
           validatorSet.connect(this.signers.governance).removeFromWhitelist([this.signers.validators[3].address])
         ).to.not.be.reverted;
 
-        expect((await validatorSet.validators(this.signers.validators[1].address)).whitelisted).to.be.false;
+        expect((await validatorSet.validators(this.signers.validators[3].address)).status).to.be.equal(
+          VALIDATOR_STATUS.None
+        );
+
+        expect((await validatorSet.validators(this.signers.validators[1].address)).status).to.be.equal(
+          VALIDATOR_STATUS.Whitelisted
+        );
+      });
+
+      it("should revert if we remove from whitelist twice", async function () {
+        const { validatorSet } = await loadFixture(this.fixtures.initializedValidatorSetStateFixture);
+
+        await expect(
+          validatorSet
+            .connect(this.signers.governance)
+            .addToWhitelist([this.signers.validators[1].address, this.signers.validators[3].address])
+        ).to.not.be.reverted;
+
+        await expect(
+          validatorSet.connect(this.signers.governance).removeFromWhitelist([this.signers.validators[3].address])
+        ).to.not.be.reverted;
+
+        await expect(
+          validatorSet.connect(this.signers.governance).removeFromWhitelist([this.signers.validators[3].address])
+        ).to.be.revertedWith("Must be whitelisted.");
+
+        expect((await validatorSet.validators(this.signers.validators[3].address)).status).to.be.equal(
+          VALIDATOR_STATUS.None
+        );
+
+        expect((await validatorSet.validators(this.signers.validators[1].address)).status).to.be.equal(
+          VALIDATOR_STATUS.Whitelisted
+        );
+      });
+
+      it("should not be able to whitelist user that is with status Registered/Banned or already Whitelisted", async function () {
```

Comment:
You can just use preset fixtures of staked and registered validators to make this test instead of registering a new validator

Reply:
simplify the test so it gets a registered user from fixtures